tag/download_vott_json.py

```diff
@@ -44,24 +49,43 @@ def make_vott_output(all_predictions, output_location, user_folders, image_loc,
     for i, predictions in enumerate(all_predictions):
         all_frames = []
         set_predictions = defaultdict(list)
-        for prediction in predictions:
-            set_predictions[tuple(map(float,prediction[2:8]))].append(prediction[1])
+        if max_tags_per_pixel is None:
+            for prediction in predictions:
+                x_1, x_2, y_1, y_2, height, width = map(float, prediction[2:8])
+                if prediction[TAG_LOCATION]!="NULL" and (x_1,x_2,y_1,y_2)!=(0,0,0,0):
+                    x_1 = int(x_1*width)
+                    x_2 = int(x_2*width)
+                    y_1 = int(y_1*height)
+                    y_2 = int(y_2*height)
+                    set_predictions[(x_1, x_2, y_1, y_2, height, width)].append(prediction[TAG_LOCATION])
+        else:
+            if predictions:
+                num_tags = np.zeros((int(predictions[0][6]),int(predictions[0][7])), dtype=int)
```

Comment:
Similar here: you have  assumptions about array dim and structure. What is predictions[0][6], predictions[0][7]?
It quite hard to read\maintain such code :)